src/main/webapp/scripts/session-script.js

```diff
@@ -40,9 +70,16 @@ window.onload = function() { main(); }
 function main() {
   urlParameters = new URLSearchParams(window.location.search);
   serverClient = new ServerClient(urlParameters);
+  noVNCClient = new NoVNCClient(
+      connectCallback, disconnectCallback, 
+          document.getElementById('session-screen'));
   addOnClickListenerToElements();
   serverClient.getSession().then(session => {
+    noVNCClient.remoteToSession(session.getIpOfVM(), 
+        session.getSessionId());
     setReadOnlyInputs(session.getSessionId());
+    document.getElementById('welcome-message').style.display = 'block';
+    updateUI();
   }).catch(error => {
     window.alert('No contact with the server!');
```

Comment:
Does a user just have to reload the page? what happens if there's no contact with the server?

Reply:
hmm, good POINT